recipes/cpr/all/conanfile.py

```diff
@@ -120,42 +140,98 @@ def _configure_cmake(self):
-                self._cmake.definitions[self._get_cmake_option("CPR_FORCE_WINSSL_BACKEND")] = self.options.get_safe("with_winssl", False)
-
-            supports_any_ssl = self.options.get_safe("with_openssl", False) or self.options.get_safe("with_winssl", False)
-            if not self._uses_old_cmake_options and not supports_any_ssl:
+
+            ssl_value = self._get_ssl_library()
+            SSL_OPTIONS = {
+                "CPR_FORCE_DARWINSSL_BACKEND": ssl_value == "darwinssl",
+                "CPR_FORCE_OPENSSL_BACKEND": ssl_value == "openssl",
+                "CPR_FORCE_WINSSL_BACKEND": ssl_value == "winssl",
+                "CMAKE_USE_OPENSSL": ssl_value == "openssl"
+            }
+
+            for cmake_option, value in SSL_OPTIONS.items():
+                self._cmake.definitions[self._get_cmake_option(cmake_option)] = value
+
+            # If we are on a version where disabling SSL requires a cmake option, disable it
+            if not self._uses_old_cmake_options and self._get_ssl_library() == CprConan._NO_SSL:
                 self._cmake.definitions["CPR_ENABLE_SSL"] = False
+
             self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
+    # Check if the system supports the given ssl library
+    def _supports_ssl_library(self, library):
+        if library == CprConan._NO_SSL:
+            return True
+        elif library == CprConan._AUTO_SSL:
+            return self._can_auto_ssl
+
+        validators = {
+            "openssl": self._supports_openssl,
+            "darwinssl": self._supports_darwinssl,
+            "winssl": self._supports_winssl
+        }
+
+        if library not in validators:
+            # This should never happen, as the options are validated by conan.
+            raise ValueError(f"Unknown SSL library: {library}")
+
+        return validators[library]
+
+    def _print_deprecation_messages(self):
+        if self.options.get_safe("with_openssl") != "deprecated":
+            self.output.warn("with_openssl is deprecated. Please use the with_ssl option.")
+        elif self.options.get_safe("with_winssl") != "deprecated":
+            self.output.warn("with_winssl is deprecated. Please use the with_ssl option.")
+
+    # Get the configured ssl library
+    def _get_ssl_library(self):
+        ssl_library = str(self.options.get_safe("with_ssl"))
+        # These must check for True so that we don't take "deprecated" to be truthy
+        if self.options.get_safe("with_openssl") == True:
+            return "openssl"
+        elif self.options.get_safe("with_winssl") == True:
+            return "winssl"
+        elif not self._can_auto_ssl and ssl_library == CprConan._AUTO_SSL:
+            if self._supports_openssl:
+                self.output.info("Auto SSL is not available below version 1.6.0. Falling back to openssl")
+                return "openssl"
```

Comment:
Perhaps this should be moved to `config_options`

such that the default is sensibly configured. We also try to no shock the user by override their explicit input and a minimum this should be a `raise ConanInvalidConfiguration`

Reply:
Makes total sense. Will fix this up later today.